common/src/main/java/com/wynntils/services/itemfilter/ItemFilter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.models.items.WynnItem;
+
+public abstract class ItemFilter {
+    protected String filterValue;
+
+    protected ItemFilter(String filterValue) {
+        this.filterValue = filterValue;
+    }
+
+    /**
+     * Prepares the filter with the provided value. Similar to compiling for a regex
+     * @throws InvalidSyntaxException if the provided value is not valid
+     */
+    public abstract void prepare() throws InvalidSyntaxException;
```

Comment:
General comment about this class, why can't you just call prepare in the constructor? That way you can remove all the "isPrepared" checks from subclasses.